File: VB/T262163/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports T262163.Reports

Namespace T262163.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function DocumentViewerPartial() As ActionResult
			Dim report As New ProductsReport()
			Return PartialView("_DocumentViewerPartial", report)
		End Function

		Public Function DocumentViewerPartialExport() As ActionResult
			Dim report As New ProductsReport()
			Return DocumentViewerExtension.ExportTo(report, Request)
		End Function
	End Class
End Namespace

File: CS/T262163/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using T262163.Reports;

namespace T262163.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult DocumentViewerPartial() {
            ProductsReport report = new ProductsReport();
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerPartialExport() {
            ProductsReport report = new ProductsReport();
            return DocumentViewerExtension.ExportTo(report, Request);
        }
    }
}